pkg/reconcilers/base_reconciler.go

```diff
@@ -0,0 +1,113 @@
+package reconcilers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/3scale/apicast-operator/pkg/k8sutils"
+)
+
+// MutateFn is a function which mutates the existing object into it's desired state.
+type MutateFn func(existing, desired k8sutils.KubernetesObject) (bool, error)
```

Comment:
what MutateFn takes is a k8sutils.KubernetesObject.
It usually happens that you are interested on specific fields of a specific Kind.

I understand then that the idea (as I've seen you've done in some of them) is that the MutateFn methods that we implement should contain a type assertion to check and convert to the desired type right?

Reply:
Yes. It is a toll to be paid. There is no other way, AFAIK. Golang requires it.

Reply:
Yes that's what I thought. Just wanted to make sure I understood it for future implementations of those methods :+1: